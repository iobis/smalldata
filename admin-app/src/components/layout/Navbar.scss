.navbar {
  margin-bottom: var(--margin-bottom);
  &.is-info {
    background-color: var(--admin-color);
  }

  .navbar-menu .navbar-start a {
    font-family: var(--title-font);

    &.navbar-item.is-active,
    &.navbar-item:hover {
      background-color: var(--obis-green);
    }
  }

  .navbar-brand {
    p {
      padding-right: 10px;
      font-size: 26px;
      font-weight: bold;
    }
  }

  .navbar-brand > .navbar-item.logo {
    background-color: white;
    color: var(--font-color);
    img {
      margin-right: 0.5rem;
    }
  }
}

.navbar.is-info .navbar-brand {
  a.navbar-item.logo:hover {
    background-color: transparent;
  }
}

.navbar.is-info .login-nav-item.navbar-end {
  .navbar-item {
    &.auth-button:hover {
      background-color: var(--obis-green);
    }
  }
}
