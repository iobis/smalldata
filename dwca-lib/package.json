{
  "name": "@smalldata/dwca-lib",
  "version": "1.0.0",
  "private": true,
  "main": "src/index.js",
  "scripts": {
    "lint": "echo Nothig to lint yet",
    "test:ci": "jest --passWithNoTests"
  },
  "dependencies": {
    "react": "16.8.6"
  },
  "devDependencies": {
    "@babel/core": "7.2.2",
    "@babel/preset-env": "7.3.1",
    "@babel/preset-react": "7.0.0",
    "babel-jest": "24.0.0",
    "jest": "24.7.1",
    "react-test-renderer": "16.8.5"
  }
}
