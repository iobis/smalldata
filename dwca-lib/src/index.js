import Button from './Button/Button'

function log(name) {
  console.log('log:', name)
}

export default {
  Button,
  log
}
