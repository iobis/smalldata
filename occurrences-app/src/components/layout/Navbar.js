/* eslint-disable jsx-a11y/anchor-is-valid */
import classNames from 'classnames'
import React, { useRef, useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link, NavLink } from 'react-router-dom'
import { useOnClickOutside } from '../../hooks/hooks'
import { useTranslation } from 'react-i18next'

export default function Navbar() {
  const { t } = useTranslation()
  const [navbarMenuActive, setNavbarMenuActive] = useState(false)
  const menuRef = useRef()
  const hideNavbarMenu = () => setNavbarMenuActive(false)
  useOnClickOutside(menuRef, hideNavbarMenu)

  return (
    <nav aria-label="main navigation" className="navbar is-info" ref={menuRef} role="navigation">
      <div className="navbar-brand">
        <Link className="navbar-item" onClick={hideNavbarMenu} to="/input-data">
          <p style={{ 'width': 112, 'fontSize': 26, 'fontWeight': 'bold' }}>OBIS</p>
        </Link>
        <a
          className="navbar-burger"
          onClick={() => setNavbarMenuActive(!navbarMenuActive)}
          role="button">
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>
      <div className={classNames('navbar-menu', { 'is-active': navbarMenuActive })}>
        <div className="navbar-start">
          <NavbarItem onClick={hideNavbarMenu} to="/input-data">
            {t('navbar.inputData')}
          </NavbarItem>
          <NavbarItem onClick={hideNavbarMenu} to="/help">
            {t('navbar.help')}
          </NavbarItem>
        </div>
        <div className="navbar-end">
          <a className="navbar-item" onClick={() => console.log('TBD: logout clicked')}>
            <span className="icon" style={{ 'marginRight': 6 }}><FontAwesomeIcon icon="user"/></span>
            {t('navbar.logout')}
          </a>
        </div>
      </div>
    </nav>
  )
}

const NavbarItem = ({ children, onClick, to }) => (
  <NavLink activeClassName="is-active" className="navbar-item" onClick={onClick} to={to}>
    {children}
  </NavLink>
)
