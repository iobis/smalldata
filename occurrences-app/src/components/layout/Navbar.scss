.navbar {
  &.is-info {
    background-color: transparent;
  }

  .navbar-menu .navbar-start a {
    font-family: var(--title-font);

    &.navbar-item {
      color: var(--font-color);
    }

    &.navbar-item.is-active,
    &.navbar-item:hover {
      color: white;
      background-color: var(--obis-green);
    }
  }

  .navbar-brand {
    p {
      padding-right: 10px;
      font-size: 26px;
      font-weight: bold;
    }

    a.navbar-item.logo {
      background-color: white;
      color: var(--font-color);

      &:hover,
      &:visited {
        background-color: white;
        color: var(--font-color);
      }
    }
  }

  .login-nav-item.navbar-end {
    .navbar-item {
      color: var(--font-color);
      &.auth-button:hover {
        background-color: var(--obis-green);
        color: white;
      }
    }
  }

  .navbar-burger {
    span {
      color: black;
    }
  }
}
