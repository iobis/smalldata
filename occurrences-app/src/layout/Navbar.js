import classNames from 'classnames'
import React, { useRef, useState } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link, NavLink } from 'react-router-dom'
import { useOnClickOutside } from '../hooks/hooks'
import { useTranslation } from 'react-i18next'

export default function Navbar() {
  const { t } = useTranslation()
  const [navbarMenuActive, setNavbarMenuActive] = useState(false)
  const menuRef = useRef()

  const onNavbarItemClick = () => setNavbarMenuActive(false)

  useOnClickOutside(menuRef, () => setNavbarMenuActive(false))

  return (
    <nav className="navbar is-info" role="navigation" ref={menuRef} aria-label="main navigation">
      <div className="navbar-brand">
        <Link to="/input-data" className="navbar-item">
          <p style={{ 'width': 112, 'fontSize': 26, 'fontWeight': 'bold' }}>OBIS</p>
        </Link>
        <a
          role="button"
          className="navbar-burger"
          onClick={() => setNavbarMenuActive(!navbarMenuActive)}>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>
      <div className={classNames('navbar-menu', { 'is-active': navbarMenuActive })}>
        <div className="navbar-start">
          <NavbarItem onClick={onNavbarItemClick} to="/input-data">
            {t('navbar.inputData')}
          </NavbarItem>
          <NavbarItem onClick={onNavbarItemClick} to="/help">
            {t('navbar.help')}
          </NavbarItem>
        </div>
        <div className="navbar-end">
          <a className="navbar-item" onClick={() => console.log('TBD: logout clicked')}>
            <span className="icon" style={{ 'marginRight': 6 }}><FontAwesomeIcon icon="user"/></span>
            {t('navbar.logout')}
          </a>
        </div>
      </div>
    </nav>
  )
}

const NavbarItem = ({ children, onClick, to }) => (
  <NavLink activeClassName="is-active" className="navbar-item" to={to} onClick={onClick}>
    {children}
  </NavLink>
)
