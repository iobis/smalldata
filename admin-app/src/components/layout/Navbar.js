/* eslint-disable jsx-a11y/anchor-is-valid */
import classNames from 'classnames'
import PropTypes from 'prop-types'
import React, { useContext, useRef, useState } from 'react'
import { AuthContext } from '../../context/AuthContext'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { hooks } from '@smalldata/dwca-lib'
import { Link, NavLink } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

export default function Navbar() {
  const { t } = useTranslation()
  const { loggedIn } = useContext(AuthContext)
  const [navbarMenuActive, setNavbarMenuActive] = useState(false)
  const menuRef = useRef()
  const hideNavbarMenu = () => setNavbarMenuActive(false)
  hooks.useOnClickOutside(menuRef, hideNavbarMenu)

  return (
    <nav aria-label="main navigation" className="navbar is-info" ref={menuRef} role="navigation">
      <div className="navbar-brand">
        <Link className="navbar-item" onClick={hideNavbarMenu} to="/manage-dataset">
          <p>OBIS</p>
        </Link>
        <a
          className="navbar-burger"
          onClick={() => setNavbarMenuActive(!navbarMenuActive)}
          role="button">
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>
      <div className={classNames('navbar-menu', { 'is-active': navbarMenuActive })}>
        {loggedIn ? (
          <div className="navbar-start">
            <NavbarItem onClick={hideNavbarMenu} to="/manage-dataset">
              {t('navbar.manageDataset')}
            </NavbarItem>
            <NavbarItem onClick={hideNavbarMenu} to="/manage-users">
              {t('navbar.manageUsers')}
            </NavbarItem>
            <NavbarItem onClick={hideNavbarMenu} to="/input-data">
              {t('navbar.inputData')}
            </NavbarItem>
          </div>
        ) : null}
        <LoginNavItem loggedIn={loggedIn}/>
      </div>
    </nav>
  )
}

function LoginNavItem({ loggedIn }) {
  const { claims, logOut, redirectToOceanExpert } = useContext(AuthContext)
  const name = claims && claims.name

  return (
    <div className="login-nav-item navbar-end">
      <span className="navbar-item">{name}</span>
      {loggedIn
        ? <AuthButton labelKey="navbar.logout" onClick={logOut}/>
        : <AuthButton labelKey="navbar.login" onClick={redirectToOceanExpert}/>}
    </div>
  )
}

LoginNavItem.propTypes = {
  loggedIn: PropTypes.bool.isRequired
}

function AuthButton({ labelKey, onClick }) {
  const { t } = useTranslation()

  return (
    <a className="auth-button navbar-item" onClick={onClick}>
      <span className="icon" style={{ 'marginRight': 6 }}>
        <FontAwesomeIcon icon="user"/>
      </span>
      {t(labelKey)}
    </a>
  )
}

AuthButton.propTypes = {
  labelKey: PropTypes.string.isRequired,
  onClick:  PropTypes.func.isRequired
}

const NavbarItem = ({ children, onClick, to }) => (
  <NavLink activeClassName="is-active" className="navbar-item" onClick={onClick} to={to}>
    {children}
  </NavLink>
)

NavbarItem.propTypes = {
  children: PropTypes.string.isRequired,
  onClick:  PropTypes.func.isRequired,
  to:       PropTypes.string.isRequired
}
