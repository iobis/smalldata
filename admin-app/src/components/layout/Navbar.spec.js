import Navbar from './Navbar'
import React from 'react'
import { AuthContext } from '@smalldata/dwca-lib'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'

describe('Navbar', () => {
  let authProviderValue
  let logOut
  let redirectToOceanExpert
  let wrapper

  describe('when logged in as "node manager"', () => {
    it('renders correctly for route / ', () => {
      wrapper = mount(
        <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
          <AuthContext.Provider value={createAuthProviderValue()}>
            <Navbar/>
          </AuthContext.Provider>
        </MemoryRouter>
      )

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.navbar-start a.navbar-item')).toHaveLength(2)
      wrapper.find('.login-nav-item .auth-button').simulate('click')
      expect(logOut).toHaveBeenCalledTimes(1)
      expect(redirectToOceanExpert).toHaveBeenCalledTimes(0)
    })

    it('renders correctly for route /manage-dataset ', () => {
      wrapper = mount(
        <MemoryRouter initialEntries={[{ pathname: '/manage-dataset', key: 'testKey' }]}>
          <AuthContext.Provider value={createAuthProviderValue()}>
            <Navbar/>
          </AuthContext.Provider>
        </MemoryRouter>
      )

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.navbar-start a.navbar-item')).toHaveLength(2)
    })

    it('renders correctly for route /manage-users ', () => {
      wrapper = mount(
        <MemoryRouter initialEntries={[{ pathname: '/manage-users', key: 'testKey' }]}>
          <AuthContext.Provider value={createAuthProviderValue()}>
            <Navbar/>
          </AuthContext.Provider>
        </MemoryRouter>
      )

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.navbar-start a.navbar-item')).toHaveLength(2)
    })
  })

  it('renders correctly when logged out', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <AuthContext.Provider value={createAuthProviderValue({ loggedIn: false })}>
          <Navbar/>
        </AuthContext.Provider>
      </MemoryRouter>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.navbar-start a.navbar-item')).toHaveLength(0)
    wrapper.find('.login-nav-item .auth-button').simulate('click')
    expect(redirectToOceanExpert).toHaveBeenCalledTimes(1)
  })

  it('renders correctly when logged in as "researcher"', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <AuthContext.Provider value={createAuthProviderValue({ role: 'researcher' })}>
          <Navbar/>
        </AuthContext.Provider>
      </MemoryRouter>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.navbar-start a.navbar-item')).toHaveLength(0)
    wrapper.find('.login-nav-item .auth-button').simulate('click')
    expect(logOut).toHaveBeenCalledTimes(1)
    expect(redirectToOceanExpert).toHaveBeenCalledTimes(0)
  })

  function createAuthProviderValue(props) {
    logOut = jest.fn()
    redirectToOceanExpert = jest.fn()
    authProviderValue = {
      claims:   { name: 'Charles Darwin' },
      role:     'node manager',
      loggedIn: true,
      logOut,
      redirectToOceanExpert,
      userRef:  'ovZTtaOJZ98xDDY'
    }
    return { ...authProviderValue, ...props }
  }
})
