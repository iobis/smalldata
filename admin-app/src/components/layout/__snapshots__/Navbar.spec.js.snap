// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly for route / 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AuthProvider>
      <Navbar>
        <nav
          aria-label="main navigation"
          className="navbar is-info"
          role="navigation"
        >
          <div
            className="navbar-brand"
          >
            <Link
              className="navbar-item"
              onClick={[Function]}
              to="/manage-dataset"
            >
              <a
                className="navbar-item"
                href="/manage-dataset"
                onClick={[Function]}
              >
                <p>
                  OBIS
                </p>
              </a>
            </Link>
            <a
              className="navbar-burger"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
            </a>
          </div>
          <div
            className="navbar-menu"
          >
            <LoginNavItem
              loggedIn={false}
            >
              <div
                className="login-nav-item navbar-end"
              >
                <span
                  className="navbar-item"
                />
                <AuthButton
                  labelKey="navbar.login"
                  onClick={[Function]}
                >
                  <a
                    className="auth-button navbar-item"
                    onClick={[Function]}
                  >
                    <span
                      className="icon"
                      style={
                        Object {
                          "marginRight": 6,
                        }
                      }
                    >
                      <FontAwesomeIcon
                        icon="user"
                      >
                        <i
                          className="fa"
                          icon="user"
                        />
                      </FontAwesomeIcon>
                    </span>
                    navbar.login
                  </a>
                </AuthButton>
              </div>
            </LoginNavItem>
          </div>
        </nav>
      </Navbar>
    </AuthProvider>
  </Router>
</MemoryRouter>
`;

exports[`Navbar renders correctly for route /help 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/help",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/help",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/help",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AuthProvider>
      <Navbar>
        <nav
          aria-label="main navigation"
          className="navbar is-info"
          role="navigation"
        >
          <div
            className="navbar-brand"
          >
            <Link
              className="navbar-item"
              onClick={[Function]}
              to="/manage-dataset"
            >
              <a
                className="navbar-item"
                href="/manage-dataset"
                onClick={[Function]}
              >
                <p>
                  OBIS
                </p>
              </a>
            </Link>
            <a
              className="navbar-burger"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
            </a>
          </div>
          <div
            className="navbar-menu"
          >
            <LoginNavItem
              loggedIn={false}
            >
              <div
                className="login-nav-item navbar-end"
              >
                <span
                  className="navbar-item"
                />
                <AuthButton
                  labelKey="navbar.login"
                  onClick={[Function]}
                >
                  <a
                    className="auth-button navbar-item"
                    onClick={[Function]}
                  >
                    <span
                      className="icon"
                      style={
                        Object {
                          "marginRight": 6,
                        }
                      }
                    >
                      <FontAwesomeIcon
                        icon="user"
                      >
                        <i
                          className="fa"
                          icon="user"
                        />
                      </FontAwesomeIcon>
                    </span>
                    navbar.login
                  </a>
                </AuthButton>
              </div>
            </LoginNavItem>
          </div>
        </nav>
      </Navbar>
    </AuthProvider>
  </Router>
</MemoryRouter>
`;

exports[`Navbar renders correctly for route /input-data 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/input-data",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/input-data",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/input-data",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AuthProvider>
      <Navbar>
        <nav
          aria-label="main navigation"
          className="navbar is-info"
          role="navigation"
        >
          <div
            className="navbar-brand"
          >
            <Link
              className="navbar-item"
              onClick={[Function]}
              to="/manage-dataset"
            >
              <a
                className="navbar-item"
                href="/manage-dataset"
                onClick={[Function]}
              >
                <p>
                  OBIS
                </p>
              </a>
            </Link>
            <a
              className="navbar-burger"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
            </a>
          </div>
          <div
            className="navbar-menu"
          >
            <LoginNavItem
              loggedIn={false}
            >
              <div
                className="login-nav-item navbar-end"
              >
                <span
                  className="navbar-item"
                />
                <AuthButton
                  labelKey="navbar.login"
                  onClick={[Function]}
                >
                  <a
                    className="auth-button navbar-item"
                    onClick={[Function]}
                  >
                    <span
                      className="icon"
                      style={
                        Object {
                          "marginRight": 6,
                        }
                      }
                    >
                      <FontAwesomeIcon
                        icon="user"
                      >
                        <i
                          className="fa"
                          icon="user"
                        />
                      </FontAwesomeIcon>
                    </span>
                    navbar.login
                  </a>
                </AuthButton>
              </div>
            </LoginNavItem>
          </div>
        </nav>
      </Navbar>
    </AuthProvider>
  </Router>
</MemoryRouter>
`;

exports[`Navbar renders correctly for route /input-data/new 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/input-data/new",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/input-data/new",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/input-data/new",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AuthProvider>
      <Navbar>
        <nav
          aria-label="main navigation"
          className="navbar is-info"
          role="navigation"
        >
          <div
            className="navbar-brand"
          >
            <Link
              className="navbar-item"
              onClick={[Function]}
              to="/manage-dataset"
            >
              <a
                className="navbar-item"
                href="/manage-dataset"
                onClick={[Function]}
              >
                <p>
                  OBIS
                </p>
              </a>
            </Link>
            <a
              className="navbar-burger"
              onClick={[Function]}
              role="button"
            >
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
              <span
                aria-hidden="true"
              />
            </a>
          </div>
          <div
            className="navbar-menu"
          >
            <LoginNavItem
              loggedIn={false}
            >
              <div
                className="login-nav-item navbar-end"
              >
                <span
                  className="navbar-item"
                />
                <AuthButton
                  labelKey="navbar.login"
                  onClick={[Function]}
                >
                  <a
                    className="auth-button navbar-item"
                    onClick={[Function]}
                  >
                    <span
                      className="icon"
                      style={
                        Object {
                          "marginRight": 6,
                        }
                      }
                    >
                      <FontAwesomeIcon
                        icon="user"
                      >
                        <i
                          className="fa"
                          icon="user"
                        />
                      </FontAwesomeIcon>
                    </span>
                    navbar.login
                  </a>
                </AuthButton>
              </div>
            </LoginNavItem>
          </div>
        </nav>
      </Navbar>
    </AuthProvider>
  </Router>
</MemoryRouter>
`;

exports[`Navbar renders correctly when logged in 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar>
      <nav
        aria-label="main navigation"
        className="navbar is-info"
        role="navigation"
      >
        <div
          className="navbar-brand"
        >
          <Link
            className="navbar-item"
            onClick={[Function]}
            to="/manage-dataset"
          >
            <a
              className="navbar-item"
              href="/manage-dataset"
              onClick={[Function]}
            >
              <p>
                OBIS
              </p>
            </a>
          </Link>
          <a
            className="navbar-burger"
            onClick={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </a>
        </div>
        <div
          className="navbar-menu"
        >
          <div
            className="navbar-start"
          >
            <NavbarItem
              onClick={[Function]}
              to="/manage-dataset"
            >
              <NavLink
                activeClassName="is-active"
                className="navbar-item"
                onClick={[Function]}
                to="/manage-dataset"
              >
                <Route
                  path="\\\\/manage-dataset"
                >
                  <Link
                    aria-current={null}
                    className="navbar-item"
                    onClick={[Function]}
                    to="/manage-dataset"
                  >
                    <a
                      aria-current={null}
                      className="navbar-item"
                      href="/manage-dataset"
                      onClick={[Function]}
                    >
                      navbar.manageDataset
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </NavbarItem>
            <NavbarItem
              onClick={[Function]}
              to="/manage-users"
            >
              <NavLink
                activeClassName="is-active"
                className="navbar-item"
                onClick={[Function]}
                to="/manage-users"
              >
                <Route
                  path="\\\\/manage-users"
                >
                  <Link
                    aria-current={null}
                    className="navbar-item"
                    onClick={[Function]}
                    to="/manage-users"
                  >
                    <a
                      aria-current={null}
                      className="navbar-item"
                      href="/manage-users"
                      onClick={[Function]}
                    >
                      navbar.manageUsers
                    </a>
                  </Link>
                </Route>
              </NavLink>
            </NavbarItem>
          </div>
          <LoginNavItem
            loggedIn={true}
          >
            <div
              className="login-nav-item navbar-end"
            >
              <span
                className="navbar-item"
              >
                Zeus
              </span>
              <AuthButton
                labelKey="navbar.logout"
                onClick={[MockFunction]}
              >
                <a
                  className="auth-button navbar-item"
                  onClick={[MockFunction]}
                >
                  <span
                    className="icon"
                    style={
                      Object {
                        "marginRight": 6,
                      }
                    }
                  >
                    <FontAwesomeIcon
                      icon="user"
                    >
                      <i
                        className="fa"
                        icon="user"
                      />
                    </FontAwesomeIcon>
                  </span>
                  navbar.logout
                </a>
              </AuthButton>
            </div>
          </LoginNavItem>
        </div>
      </nav>
    </Navbar>
  </Router>
</MemoryRouter>
`;

exports[`Navbar renders correctly when logged out 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navbar>
      <nav
        aria-label="main navigation"
        className="navbar is-info"
        role="navigation"
      >
        <div
          className="navbar-brand"
        >
          <Link
            className="navbar-item"
            onClick={[Function]}
            to="/manage-dataset"
          >
            <a
              className="navbar-item"
              href="/manage-dataset"
              onClick={[Function]}
            >
              <p>
                OBIS
              </p>
            </a>
          </Link>
          <a
            className="navbar-burger"
            onClick={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </a>
        </div>
        <div
          className="navbar-menu"
        >
          <LoginNavItem
            loggedIn={false}
          >
            <div
              className="login-nav-item navbar-end"
            >
              <span
                className="navbar-item"
              />
              <AuthButton
                labelKey="navbar.login"
                onClick={[MockFunction]}
              >
                <a
                  className="auth-button navbar-item"
                  onClick={[MockFunction]}
                >
                  <span
                    className="icon"
                    style={
                      Object {
                        "marginRight": 6,
                      }
                    }
                  >
                    <FontAwesomeIcon
                      icon="user"
                    >
                      <i
                        className="fa"
                        icon="user"
                      />
                    </FontAwesomeIcon>
                  </span>
                  navbar.login
                </a>
              </AuthButton>
            </div>
          </LoginNavItem>
        </div>
      </nav>
    </Navbar>
  </Router>
</MemoryRouter>
`;
